src/PAModel/CanvasDocument.cs

```diff
@@ -390,6 +404,15 @@ internal void ApplyBeforeMsAppWriteTransforms(ErrorContainer errors)
             // Also add Screen and App templates (not xml, constructed in code on the server)
             GlobalTemplates.AddCodeOnlyTemplates(_templateStore, templateDefaults, _properties.DocumentAppType);
 
+            // Generate DynamicControlDefinitionJson for power apps controls
+            foreach (var kvp in _powerAppsControls)
+            {
+                if (_templateStore.TryGetTemplate(kvp.Key, out var template))
```

Comment:
Should there be an error/warning if the template isn't found?

Reply:
We do validate if the template was deleted in ValidateIfTemplateExists method